public/app/themes/justice/inc/documents.php

```diff
@@ -296,4 +360,89 @@ public function addRewriteRules($rules)
 
         return array_merge($new_rules, $rules);
     }
+
+    /**
+     * Functions related to documents, also having an effect on non-documents.
+     * - isValidSlug
+     * - mediaLibraryHint
+     * - removeFileSupport
+     * - setUploadSizeLimit
+     */
+
+
+    /**
+     * isValidSlug
+     * Prevent all single posts, of any post type, from using the document(s) slug.
+     * This prevents a conflict, where nested pages may make a url like /documents/my-document
+     */
+
```

Comment:
@EarthlingDavey this is more of an observation than an issue, but is it worth using PHPDoc comments to describe functions? Probably more for a team discussion around code standards generally.

```
/**
* Prevent all single posts, of any post type, from using the document(s) slug.
*
* @param bool $bad_slug Whether the slug is bad or not
* @param string $slug The document slug
* @return bool The sum of the two numbers.
*/
```

https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/